rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -49,6 +49,30 @@ fn generate_timeseries_id(
     adler.checksum()
 }
 
+// MetricType should be used only with PartialMetricsMessage. Its purpose is to
+// be able to categorize the type of metric message we see without incurring the
+// full cost of deserializing the entire payload.
+#[derive(Debug, Deserialize, PartialEq)]
+#[serde(tag = "type")]
```

Comment:
This `tag` along with the `flatten` below looks very weird, but I was checking the docs and it seems like this is the way.
Otherwise, we usually serialize such data-less enums using `FromStr` and `Display`, with some annotations of how to serialize them. Though I would say this approach here is fine as well.

Reply:
Yeah this is definitely more nuanced than other fields which need serialization/deserialization